reVX/wind_dirs/prominent_wind_dirs.py

```diff
@@ -0,0 +1,244 @@
+                                 chunk_point_len=chunk_point_len)
+
+        meta = agg_out.pop('meta')
+        neighbor_gids = self._get_neighbors(self._excl_fpath,
+                                            meta['sc_point_gids'].values,
+                                            resolution=self._resolution)
+
+        dir_pos = self._map_direction_pos(self._h5_fpath)
+        prominent_dirs = agg_out.pop('powerrose_100m')[dir_pos]
+        prominent_dirs = np.argsort(prominent_dirs.T)
+        prominent_dirs = np.take_along_axis(neighbor_gids.values,
+                                            prominent_dirs,
+                                            axis=1)
+
+        columns = ['prominent_direction_{}'.format(i + 1)
+                   for i in range(len(dir_pos))]
+        prominent_dirs = pd.DataFrame(prominent_dirs,
+                                      index=meta['sc_point_gids'].values,
+                                      columns=columns)
+        prominent_dirs = neighbor_gids.join(prominent_dirs)
+        del neighbor_gids
+        prominent_dirs = meta.join(prominent_dirs)
+
+        return prominent_dirs
+
+    @classmethod
+    def run(cls, power_rose_h5_fpath, excl_fpath,
+            agg_dset='powerrose_100m', tm_dset='techmap_wtk',
+            resolution=64, excl_area=0.0081, max_workers=None,
+            chunk_point_len=100, out_fpath=None):
+        """
+        Aggregate powerrose to supply curve points, find neighboring supply
+        curve point gids and rank them based on prominent powerrose direction
+
+        Parameters
+        ----------
+        power_rose_h5_fpath : str
+            Filepath to .h5 file containing powerrose data
+        excl_fpath : str
+            Filepath to exclusions h5 with techmap dataset.
+        agg_dset : str, optional
+            Powerrose dataset to aggreate, by default 'powerrose_100m'
+        tm_dset : str, optional
+            Dataset name in the techmap file containing the
+            exclusions-to-resource mapping data,
+            by default 'techmap_wtk'
+        resolution : int, optional
+            SC resolution, must be input in combination with gid. Prefered
+            option is to use the row/col slices to define the SC point instead,
+            by default 64
+        excl_area : float, optional
+            Area of an exclusion cell (square km), by default 0.0081
+        max_workers : int | None, optional
+            Number of cores to run summary on. None is all
+            available cpus, by default None
+        chunk_point_len : int, optional
+            Number of SC points to process on a single parallel worker,
+            by default 100
+        out_fpath : str
+            Path to .csv file to save output table to
+
+        Returns
+        -------
+        prominent_dirs : pandas.DataFrame
+            Update meta data table with neighboring supply curve point gids
```

Comment:
For me at least this is kind of nebulous. How are the SC point gids stored? Are there 8 new columns, one for each direction? What is the order of directions (see other comment on direction attr).

Reply:
tried to be more verbose, what I am currently doing is adding 16 new columns:
8 = the cardinal direction of each neighbor and its gid
8 = the same gids sorted in terms of prominent power rose direction

while these are redundant I thought it might be nice to have both